applications/GeoMechanicsApplication/custom_utilities/math_utilities.h

```diff
@@ -36,6 +36,9 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) GeoMechanicsMathUtilities
         return rVector / length;
     }
 
+    [[nodiscard]] static Matrix VectorToDiagonalMatrix(const Vector& rVector);
+    [[nodiscard]] static Matrix RotateTensor(const Matrix& rTensor, const Matrix& rRotationMatrix);
```

Comment:
Perhaps we should make this function name slightly more specific:
```suggestion
    [[nodiscard]] static Matrix RotateSecondOrderTensor(const Matrix& rTensor, const Matrix& rRotationMatrix);
```